### PR TITLE
add LogBidSurpassed + test, tweak publicizer to use test Address for …

### PR DESCRIPTION
**Pull request by shine2lay**
…ESCAPE_HATCH_ENABLER

### CODE REVIEW

File: contracts/ROSCA.sol [hunk 1]
```diff
@@ -39,15 +39,16 @@ contract ROSCA {
...
 
   /////////
   // EVENTS
   /////////
   event LogContributionMade(address user, uint256 amount, uint256 currentRound);
   event LogStartOfRound(uint256 currentRound);
+  event LogBidSurpassed(uint256 prevBid, address prevWinnerAddress, uint256 currentRound);
   event LogNewLowestBid(uint256 bid, address winnerAddress, uint256 currentRound);
-  event LogRoundFundsReleased(address winnerAddress, uint256 amount, uint256 currentRound);
+  event LogRoundFundsReleased(address winnerAddress, uint256 amount, uint256 currentRound, uint256 roundDiscount);
```
Comment (sihoang) [inline contracts/ROSCA.sol]: just to be consistent that currentRound is always the last parameter

File: contracts/ROSCA.sol [hunk 2]
```diff
@@ -410,6 +412,10 @@ contract ROSCA {
       LogUnsuccessfulBid(msg.sender, bid, lowestBid, currentRound);
       return;
     }
+    if (winnerAddress != 0) {
+      LogBidSurpassed(lowestBid, winnerAddress, currentRound);
```
Comment (sihoang) [inline contracts/ROSCA.sol:416]: this should be `msg.sender` and no need to be inside the `if`

Comment (sihoang) [reply contracts/ROSCA.sol:416]: nvm, please ignore. This is for the previous winner

File: test/bidUnitTest.js [hunk 3]
```diff
@@ -47,6 +47,34 @@ contract('ROSCA bid Unit Test', function(accounts) {
             "expected placing bid less than MIN_DISTRIBUTION_PERCENT threshold to throw");
     }));
 
+    it("generates a LogBidSurpassed event when a bid was outbid by another", co(function* () {
+    const BID_TO_PLACE = consts.defaultPot() * 0.94;
```
Comment (sihoang) [inline test/bidUnitTest.js]: indentation

File: contracts/ROSCA.sol (hunk 1 above)
Comment (shine2lay) [reply contracts/ROSCA.sol]: i wanted it to be consistent for currentRound to be 3rd

File: test/bidUnitTest.js (hunk 3 above)
Comment (shine2lay) [reply test/bidUnitTest.js]: done

File: contracts/ROSCA.sol (hunk 1 above)
Comment (shine2lay) [reply contracts/ROSCA.sol]: i've changed it

**Review by sihoang — APPROVED** [review]

**Pull request merged**